addons/account/tests/test_account_journal_dashboard.py

```diff
@@ -288,19 +288,25 @@ def test_gap_in_sequence_warning(self):
         moves[6].button_cancel()
         self.assertTrue(journal._query_has_sequence_holes())  # gap due to canceled move using a sequence, gap warning
 
-    def test_bank_journal_misc_operations_with_payments(self):
-        """Test that payments are excluded from the miscellaneaous operations"""
-        bank_journal = self.company_data['default_journal_bank'].copy({'currency_id': self.currency_data['currency'].id})
+    def test_bank_journal_with_default_account_as_outstanding_account_payments(self):
+        """
+        Test that payments are excluded from the miscellaneaous operations and are included in the balance
+        when having the default_account_id set as outstanding account on the journal
+        """
+        bank_journal = self.company_data['default_journal_bank'].copy()
         bank_journal.outbound_payment_method_line_ids[0].payment_account_id = bank_journal.default_account_id
         bank_journal.inbound_payment_method_line_ids[0].payment_account_id = bank_journal.default_account_id
-        self.env['account.payment'].create({
+        payment = self.env['account.payment'].create({
             'amount': 100,
             'payment_type': 'inbound',
             'partner_type': 'customer',
             'journal_id': bank_journal.id,
-        }).action_post()
+        })
+        payment.action_post()
+        payment.flush_recordset(['amount_company_currency_signed'])
```

Comment:
Flush should be done before executing the query in the business code

Reply:
Query is made when calling `_get_journal_dashboard_data_batched`, so after the flush, this or I don't understand your point here

Reply:
The flush needs to be done inside of `_get_outstanding_payments_balances`, before `cr.execute`